misc/python/materialize/cloudtest/util/environment.py

```diff
@@ -12,198 +12,201 @@
 import requests
 from requests import Response
 
-from materialize.cloudtest.config.environment_config import EnvironmentConfig
+from materialize.cloudtest.util.authentication import AuthConfig
 from materialize.cloudtest.util.common import retry
 from materialize.cloudtest.util.controller import wait_for_connectable
 from materialize.cloudtest.util.kubectl import Kubectl
 from materialize.cloudtest.util.web_request import delete, get, patch
 
 
-def create_environment_assignment(
-    config: EnvironmentConfig,
-    image: Optional[str] = None,
-) -> Dict[str, Any]:
-    environment_assignment = f"{config.auth.organization_id}-0"
-    environment = f"environment-{environment_assignment}"
-
-    json: dict[str, Any] = {}
-    if image is not None:
-        json["environmentdImageRef"] = image
-    patch(
-        config,
-        config.controllers.region_api_server.configured_base_url(),
-        "/api/region",
-        json,
-    )
-    env_kubectl = Kubectl(config.environment_context)
-    sys_kubectl = Kubectl(config.system_context)
-
-    env_kubectl.get_retry(
-        None,
-        "environment",
-        environment,
-        10,
-    )
-    return sys_kubectl.get(
-        None,
-        "environmentassignment",
-        environment_assignment,
-    )
-
-
-def wait_for_environmentd(
-    config: EnvironmentConfig, max_attempts: int = 300
-) -> Dict[str, Any]:
-    def get_environment() -> Response:
-        response = get(
-            config,
-            config.controllers.region_api_server.configured_base_url(),
+class Environment:
```

Comment:
nice! I like encapsulating all these environment management calls under this class!